src/PositionManager.sol

```diff
@@ -143,6 +153,10 @@ contract PositionManager is
         _;
     }
 
+    function tokenURI(uint256 tokenId) public view override returns (string memory) {
```

Comment:
why has this function moved files? i feel like it makes sense in the erc721 file given its part of that spec

Reply:
its because PositionDescriptor needs access to positionManager. 
so if it was in ERC721Permit_v4, it wouldn't be able to call this: 
`return IPositionDescriptor(tokenDescriptor).tokenURI(this, tokenId);`
because `this` is positionManager and erc721permit_v4 doesn't have access to that